pkg/sshd/server_test.go

```diff
@@ -194,21 +207,23 @@ func TestConcurrentPushSameRepo(t *testing.T) {
 
 // TestConcurrentPushDifferentRepo tests many concurrent pushes, each to a different repo
 func TestConcurrentPushDifferentRepo(t *testing.T) {
+	t.Skip("Need to implement an interface for git recieve to mock a fakegitreceveive")
+	t.SkipNow()
```

Comment:
haha. I started refactoring the server by adding an additional `func() error` parameter to that was responsible for executing the git-receive function. can you do that and then pass a mock one in tests?